extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/exporter/otlp/VertxGrpcExporter.java

```diff
@@ -201,20 +202,39 @@ public void handle(GrpcClientRequest<Buffer, Buffer> request) {
                 request.send(buffer).onSuccess(new Handler<>() {
                     @Override
                     public void handle(GrpcClientResponse<Buffer, Buffer> response) {
-                        GrpcStatus status = getStatus(response);
-                        if (status == GrpcStatus.OK) {
-                            exporterMetrics.addSuccess(numItems);
-                            result.succeed();
-                            return;
-                        }
-                        String statusMessage = getStatusMessage(response);
-                        if (statusMessage == null) {
-                            // TODO: this needs investigation, when this happened, the spans actually got to the server, but for some reason no status code was present in the result
-                            exporterMetrics.addSuccess(numItems);
-                            result.succeed();
-                            return;
-                        }
+                        response.exceptionHandler(new Handler<>() {
+                            @Override
+                            public void handle(Throwable t) {
+                                exporterMetrics.addFailed(numItems);
+                                logger.log(
+                                        Level.SEVERE,
+                                        "Failed to export "
+                                                + type
+                                                + "s. The response could not be parsed. Full error message: "
```

Comment:
How do you know that the response could not be parsed?

Reply:
Right... actually the stream failed, changed the error log to show that.